lib/matobjplist.gi

```diff
@@ -579,50 +657,55 @@ InstallMethod( DimensionsMat, "for a plist matrix",
 # Representation preserving constructors:
 ############################################################################
 
-InstallMethod( ZeroMatrix, "for two integers and a plist matrix",
-  [ IsInt, IsInt, IsPlistMatrixRep ],
+InstallMethod( ZeroMatrix,
+  [ "IsInt", "IsInt", "IsPlistMatrixRep" ],
   function( rows,cols,m )
-    local l,t,res;
+    local t, l;
     t := m![EMPOS];
     l := List([1..rows],i->ZeroVector(cols,t));
-    res := Objectify( TypeObj(m), [m![BDPOS],t,cols,l] );
-    if not IsMutable(m) then
-        SetFilterObj(res,IsMutable);
-    fi;
-    return res;
+    return MakeIsPlistMatrixRep( m![BDPOS], t, cols, l, false );
   end );
 
-InstallMethod( IdentityMatrix, "for an integer and a plist matrix",
-  [ IsInt, IsPlistMatrixRep ],
+InstallMethod( IdentityMatrix,
+  [ "IsInt", "IsPlistMatrixRep" ],
   function( rows,m )
-    local i,l,o,t,res;
+    local t,l,o,i;
     t := m![EMPOS];
     l := List([1..rows],i->ZeroVector(rows,t));
     o := One(m![BDPOS]);
+#T can 'o = fail' happen?
```

Comment:
Perhaps explain the motivation for this question? I assume it is because we discussed the potential use of matrices over semirings where there might be no one element (CC @james-d-mitchell ) ? 

I mean, we can't exclude the chance of this returning `fail` or triggering an error as long as we permit arbitrary BasedDomain... but then the same could be asked about zero, too? Like, the basedomain could be `PositiveIntegers`, too?

So I'd not worry too much about it, but if we feel this warrants a TODO comments, let's at least add some context to it?

Reply:
O.k., I remove this comment.
First I wanted to move it to a place that defines what an admissible base domain is, but apparently this is currently left open.